File: Codigos estudiantes por lenguaje/C++/Edwin Villarraga/matematica/A%X=B.cpp
#include<bits/stdc++.h>
#include<cstdlib>
using namespace std;
void modEquation(int a,int b){
    if(a<b){
        cout<<"no hay solucion"<<endl;
        return;
    }
    if(a==b){
        cout<<"infinitas soluciones"<<endl;
        return;
    }
    int cont=0;
    int n=a-b;
    int y=std::sqrt(a-b);

    for(int i=1;i<=y;i++){
        if(n%i==0){
            if(n/i>b){
                cont++;
                cout<<(n/i)<<" ";
            }
            if(i>b){
                cont++;
                cout<<i<<" ";
            }
        }
    }
    cout<<endl;
    if(y*y==n && y>b){
        cont--;
    }
    cout<<cont<<endl;

}
int main() {
    //a%x=b;
    int a=21,b=5;
    modEquation(a,b);

}

File: Codigos estudiantes por lenguaje/PY/Bryann Valderrama/Matemática/a%x=b.py
import math
from sys import stdout
wr = stdout.write


def modEquation(a, b):
    if a < b:
        wr(f'No hay Solucion')
        return
    if a == b:
        wr(f'Infinitas Soluciones')
    cont = 0
    n = a - b
    y = int(math.sqrt(a-b))
    for i in range(1, y+1):
        if n % i == 0:
            if n // i > b:
                cont += 1
                wr(f'{n//i}\n')
            if i > b:
                cont += 1
                wr(f'{i}\n')
    if y*y == n and y > b:
        cont -= 1
    wr(f'{cont}')


a = 21
b = 5
modEquation(a, b)
